### ISSUE TITLE
Broken with swww

### ISSUE THREAD

**Issue by cyuria** [mstoeckl/swaylock-plugin#13]
This fork seems really interesting, however when I try to lock the screen using swww, swww fails.

Here is the swww logs with `RUST_BACKTRACE=1`. This log is repeated multiple times, I'm assuming once for each process retry. The warning looks like an artifact from swww not cleaning up fully when it crashes.
```
Forked background plugin (pid = 30812): RUST_BACKTRACE=1 swww-daemon
23:34:29 [WARN] (main) socket file /run/user/1000/swww-wayland-0.socket was not deleted when the previous daemon exited
thread 'main' panicked at daemon/src/main.rs:335:14:
viewported not available: NotPresent
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: swww_daemon::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

swaylock-plugin is at the latest main commit
swww is at the release version 0.9.5 (https://github.com/LGFae/swww/tree/3ee69e3286d9aeb0614f32fe28eee4477d661b02)

The panicking file/line in question is https://github.com/LGFae/swww/blob/3ee69e3286d9aeb0614f32fe28eee4477d661b02/daemon/src/main.rs#L335

It looks to me like the `wp_viewporter` protocol isn't implemented or passed to swww correctly.

**Comment by mstoeckl** [comment]
I added support for `wp_viewporter` in 3a6ea3beb84d79dd0b945ed45251eb8bb4b02664, so `swww` might work now. (Or maybe not: when building from git main an additional patch was needed to connect, and setting images with version 0.9.5 doesn't work properly for me for some reason.)

**Comment by cyuria** [comment]
Just tested this and it looks like swww is now working.

I am unable to change the wallpaper either, however I suspect that's due to a combination of this fork requiring a separate process (i.e. unless I've missed something, it cannot use the output of a currently running process) and the way swww handles communication. Even with my custom script which ensures all environment variables are copied between the daemon and the control process, it still doesn't work. This issue is likely better raised with swww. The other option I can see to resolve this would be to find a way to "hijack" the existing swww-daemon process output, however I'm not familiar enough with wayland and other protocol extensions to know how feasible this is.

On my machine at least, the swww daemon crashes with a broken pipe error after I unlock. I'm not too sure what causes this, my suspicion is some kind of bad termination sequence, however it doesn't cause any serious issues.

Given swww works "well enough", I'll close the issue.

**Comment by cyuria** [comment]
I did a little bit of further testing and it looks like swww is unable to find it's corresponding daemon process's socket, even though it exists.

To reproduce, I executed the command as shown below in one shell and the ran `swaylock-plugin --command swww-daemon` from another.

![image](https://github.com/user-attachments/assets/0763ec20-042d-4f58-92f1-af552eae6b42)

Notice that the socket *is* created. It is unfortunately a bit hard to see because there is no separation between the two `ls` calls.
